// normal-list-react/src/normal_list.js
import React, {Component} from 'react'
import PropTypes from 'prop-types'
import './npm.css'

export default class NormalList extends Component {

    render() {
        let _props = this.props

        return (
            <div className='normal-list'>
                <Header
                    titles={_props.headerTitles}/>
                <Content
                    listData={_props.data}
                    titles={_props.headerTitles}
                    itemKeys={_props.itemKeys}
                    listClick={_props.listClick}/>
            </div>
        )
    }
}

NormalList.propTypes = {
    data: PropTypes.array.isRequired,
    headerTitles: PropTypes.array.isRequired,
    itemKeys: PropTypes.array.isRequired,
    listClick: PropTypes.func.isRequired
}

class Header extends Component {
    getItems() {
        let _titles = this.props.titles

        if(_titles) return (
            _titles.map((item, index) => {
                return (
                    <div key={index} className='normal-list-row-item'>
                        {item}
                    </div>
                )
            })
        )
    }

    render() {
        return (
            <div className='normal-list-header'>
                {this.getItems()}
            </div>
        )
    }
}

class Content extends Component {
    getItems(listIndex) {
        let _itemKeys = this.props.itemKeys
        let _data = this.props.listData[listIndex]

        if(_itemKeys) return (
            _itemKeys.map((itemKey, index) => {
                return (
                    <div key={index} className='normal-list-row-item'>
                        <span>{_data[itemKey]}</span>
                    </div>
                )
            })
        )
    }

    getList() {
        let _listData = this.props.listData

        if(_listData) return (
            _listData.map((item, index) => {
                return (
                    <div key={index} className='normal-list-row'
                        onClick={() => this.props.listClick(item)}>
                        {this.getItems(index)}
                    </div>
                )
            })
        )
    }

    render() {
        return (
            <div className='normal-list-content'>
                {this.getList()}
            </div>
        )
    }
}
